### PR TITLE
Fixes #16074 - Adds BIND variable to startup process

### PR DESCRIPTION
**Pull request by shlomizadok**
Instead of binding to 0.0.0.0 we default to localhost
and use ENV variable BIND to override.
~~This also introduces '.env' files which helps to set
other ENV variables~~

### CODE REVIEW

**Comment by shlomizadok** [comment]
/me wonders if the `.env` should be added - or explain how to use in docs

**Comment by shlomizadok** [comment]
One note: At the beginning I have added a `.env`, but on second thought it should not be added (though can be introduced in documentation)

**Comment by dLobatog** [comment]
@shlomizadok Does this work with [forklift](https://github.com/Katello/forklift)?

**Comment by shlomizadok** [comment]
Would work on forklift if we'd add a BIND variable. I'd suggest creating a `.env` via ansible (can create a pull request once this is merged)

File: Procfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Run Rails & Webpack concurrently
 # If you wish to use a different server then the default, use e.g. `export RAILS_STARTUP='puma -w 3 -p 3000 --preload'`
-rails: [ -n "$RAILS_STARTUP" ] && $RAILS_STARTUP || bin/rails server -b 0.0.0.0
+rails: [ -n "$RAILS_STARTUP" ] && $RAILS_STARTUP || [ -n "$BIND" ] && $BIND bin/rails server -b $BIND || bin/rails server -b 127.0.0.1
```
Comment (domcleal) [inline Procfile]: The `$BIND` before bin/rails appears superfluous? It causes a warning:

<pre>
13:11 $ BIND=0.0.0.0 foreman start                                                                                                                                                                                                                                              
13:11:08 rails.1   | started with pid 24145
13:11:08 webpack.1 | started with pid 24146
13:11:08 rails.1   | sh: 0.0.0.0: command not found
</pre>

**Comment by domcleal** [comment]
Please also add .env or .env\* to the gitignore file.

File: Procfile (hunk 1 above)
Comment (shlomizadok) [reply Procfile]: It was left there by accident. removed

**Comment by shlomizadok** [comment]
Added .env\* to .gitignore and removed superfluous $BIND

**Comment by domcleal** [comment]
Merged as 3b10c2a, thanks @shlomizadok.

Please update the notes at the bottom of https://theforeman.org/handbook.html#Technicalnotes on `foreman start` usage, and consider dropping an e-mail to foreman-dev in case people want to change the secure default.

**Pull request closed without merging**